### ISSUE TITLE
Error:  WorkflowSid field is required

### ISSUE THREAD

**Issue by amarkhai** [vernig/twilio-taskrouter-agent-frontend#9]
Hello! I tried to use this app as it described in Readme.md, but I had a problem with transferring incoming calls. 
When I tried to transfer a call to another agent, I saw only endless request to [http://localhost:3000/transfer?task_sid=...](url) without answer and no logs in the Node.js console (last log is **old task fetched**).
I found a couple of problem:

1. No logs here [https://github.com/vernig/twilio-taskrouter-agent-frontend/blob/master/server.js#L58](url)
    The 'create' method has 2nd argument as callback with an 'error' object as first argument. If not to pass this callback to 'create' function, you don't know is the task created or not

2.  WorkflowSid field is required
After adding this callback as a parameter and added console.log for the 'error' object, I saw the log: '**Error RestException [Error]: WorkflowSid field is required**'. As I understood correctly, now  the attribute 'workflowSid' in opts object (first argument in the create method) is required. After adding this , the call transferring started to work correctly

**Comment by vernig** [comment]
Thanks @JohnKlein1991! Let me investigate and fix that

**Comment by amarkhai** [comment]
@vernig 
And one more thing:
Do you know how to solve the problem with hanging up for calls? 
Now, if I finish a call in the browser (press 'hang up' button), this call continues on a phone.
And the same situation when I finish call on a phone  - in the browser on the /worker.html?workerSid=<Worker SID> page I see that this call is active yet

**Comment by vernig** [comment]
@JohnKlein1991 you are right, I remember it not working for this proof of concept. I reused some of the concepts from this repo into a new repo I'm working and that you may be interested in having a look: https://github.com/vernig/twilio-taskrouter-frontend. This also includes a Twilio voice client as reusable class / component. I believe I better handle call tear down there... though I'm not sure about it. I'll have to double check (i didn't have much time to work on that lately)

**Comment by vernig** [comment]
@JohnKlein1991 did you check the other repo? And does that solve your problem?

**Comment by amarkhai** [comment]
@vernig I checked it, but finally, I solved my problems after advices from Twilio support (some of them were not obvious at all). Thank you for helping!

**Comment by vernig** [comment]
Cool @JohnKlein1991 ! Thanks. Feel free to do a PR if needed. I'm not actively maintaining this solution in favour of the other one. But I'll try to add at least the hang-up

**Comment by HenBaliti** [comment]
WorkflowSid field is required :
Hey , I have got this issue when I tried to update an existing task .
So when I tried to send a "WorkflowSid" , I still got this error .
On the "TaskRouther Task Resource" on twilio docs I cant find something about this "workflowSid" on update task .

Do you know what should I do for fix that problem ?
